Feature/rout irr

BramDr opened the pull request:
Hallo Wietse,

rout_run() werkt nog niet, maar als het goed is werkt alles tot en met het sorteren van de cellen van bovenstrooms naar benedenstrooms (rout_init()).

Groetjes,
Bram

BramDr committed:
modified:   vic/drivers/image/Makefile
	new file:   vic/extensions/rout_irr/include/rout.h
	new file:   vic/extensions/rout_irr/rout.mk
	new file:   vic/extensions/rout_irr/src/rout_alloc.c
	new file:   vic/extensions/rout_irr/src/rout_finalize.c
	new file:   vic/extensions/rout_irr/src/rout_init.c
	new file:   vic/extensions/rout_irr/src/rout_run.c
	new file:   vic/extensions/rout_irr/src/rout_start.c
	new file:   vic/extensions/rout_irr/src/rout_write.c

bartnijssen committed:
Merge pull request #534 from UW-Hydro/develop

VIC 5.0.0-rc1

BramDr committed:
modified:   vic/drivers/image/src/vic_image.c
	modified:   vic/extensions/rout_irr/include/rout.h
	modified:   vic/extensions/rout_irr/src/rout_alloc.c
	modified:   vic/extensions/rout_irr/src/rout_init.c
	modified:   vic/extensions/rout_irr/src/rout_run.c
	modified:   vic/extensions/rout_irr/src/rout_start.c
	modified:   vic/extensions/rout_rvic/src/rout_init.c

wietsefranssen commented on vic/drivers/image/src/vic_image.c:
```diff
@@ -108,12 +108,12 @@ main(int    argc,
+    // allocate memory for routing
+    rout_alloc();   // Routing routine (extension)
+
     // initialize routing parameters from parameter files
```
Just curious: waarom heb je rout_alloc verplaatst?

BramDr replied:
Omdat domain_struct local_domain pas word geinitializeerd bij vic_init(),
en ik gebruik de waardes local_domain.ncells_active in de code…
On 1 Jul 2016 14:11, "Wietse Franssen" notifications@github.com wrote:

> In vic/drivers/image/src/vic_image.c
> https://github.com/wietsefranssen/VIC/pull/7#discussion_r69290001:
> 
> > ```
> >  // initialize model parameters from parameter files
> >  vic_image_init();
> > ```
> > -    // allocate memory for routing
> > -    rout_alloc();   // Routing routine (extension)
> >   +
> >    // initialize routing parameters from parameter files
> 
> Just curious: waarom heb je rout_alloc verplaatst?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/wietsefranssen/VIC/pull/7/files/b067c091c0f2ebcaaaa43aed626afc500bbcd45d#r69290001,
> or mute the thread
> https://github.com/notifications/unsubscribe/ASqP8VSd2MRd4mkn0oYAE_4XH6bV2Ew2ks5qRQPzgaJpZM4JC-E_
> .

wietsefranssen replied:
naar mijn weten wordt die al in `initialize_mpi();`bepaald...

wietsefranssen commented on vic/extensions/rout_irr/src/rout_alloc.c:
```diff
@@ -0,0 +1,166 @@
+    extern global_param_struct global_param;
+    extern domain_struct local_domain;
+    
+    rout.x_size=get_nc_dimension("/home/bram/VIC5.0Routing/VICTestSetup/input/VIC_Params/RVIC_input_NL.nc","lon");
```
Dit is onder andere de reden waarom ik het in een aparte branch wilde hebben: normaal gesproken zijn hard gecodeerde paden uit den boze in de code.
Voor het gemak en het testen snap ik waarom je het nu wel doet hoor.

wietsefranssen commented on vic/extensions/rout_irr/src/rout_alloc.c:
```diff
@@ -0,0 +1,166 @@
+    rout.x_size=get_nc_dimension("/home/bram/VIC5.0Routing/VICTestSetup/input/VIC_Params/RVIC_input_NL.nc","lon");
+    rout.y_size=get_nc_dimension("/home/bram/VIC5.0Routing/VICTestSetup/input/VIC_Params/RVIC_input_NL.nc","lat");
+    rout.total_time=global_param.nrecs; 
+    rout.uh_length=96;
```
was dit 24*4uur tijd dat het water in een rivier kan blijven?

BramDr replied:
Weet ik nog niet precies, nu staat dit nummer hier puur omdat hij ook al in de oude code werd gebruikt. Ik wil kijken of we misschien de uh per dag kunnen maken!

wietsefranssen replied:
ik was het dus ook weer kwijt. Iwan weet dit nog wel

wietsefranssen commented on vic/extensions/rout_irr/src/rout_init.c:
```diff
@@ -0,0 +1,224 @@
+        }
+    }
+        
+    set_upstream("/home/bram/VIC5.0Routing/VICTestSetup/input/VIC_Params/RVIC_input_NL.nc","flow_direction");
```
het zou op zich fijn zijn als je de paden op z'n minst relatief maakt (dan kan ik het ook testen zonder teveel in jou code te klooien)

wietsefranssen commented on vic/extensions/rout_rvic/src/rout_init.c:
```diff
@@ -95,7 +95,7 @@ rout_init(void)
                      "source_time_offset",
                      &i1start, &rout.rout_param.nSources, ivar);
-    for (i = 0; i < rout.rout_param.nSources; i++) {
+    for (i = 0; i < rout.rout_paparam_filenameram.nSources; i++) {
```
klopt dit?

BramDr replied:
hoooo, dit is een foutje :P ik zal m er de volgende keer uithalen. als het goed is verander ik niets in rout_rvic

wietsefranssen replied:
:)

wietsefranssen commented:
ziet er in het algemeen goed uit. Kan je de paden nog relatief maken en deze wijzigingen pushen (je hoeft dan niet opnieuw een merge request te sturen)?
Ik ga er aankomende week in detail naar kijken.

BramDr commented:
Momenteel niet, maar zondag of maandag denk ik wel! Is dat ok?
On 1 Jul 2016 14:23, "Wietse Franssen" notifications@github.com wrote:

> ziet er in het algemeen goed uit. Kan je de paden nog relatief maken en
> deze wijzigingen pushen (je hoeft dan niet opnieuw een merge request te
> sturen)?
> Ik ga er aankomende week in detail naar kijken.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/wietsefranssen/VIC/pull/7#issuecomment-229933322, or mute
> the thread
> https://github.com/notifications/unsubscribe/ASqP8bHXaL0egJWwTmtTHR4MxoXgegLyks5qRQazgaJpZM4JC-E_
> .

wietsefranssen commented:
prima hoor!

Commit:
Feature/driver filenames (#547)

* refactor filenames and filep structures to only include members specific to each driver, image-like drivers now only use a single PARAMETERS file

* fix bug where aggstring was not initialized in parse_output_info.c

wietsefranssen committed:
unit fix

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

BramDr committed:
modified:   vic/drivers/image/src/vic_image.c
	modified:   vic/extensions/rout_irr/include/rout.h
	modified:   vic/extensions/rout_irr/src/rout_alloc.c
	modified:   vic/extensions/rout_irr/src/rout_init.c
	modified:   vic/extensions/rout_irr/src/rout_run.c
	modified:   vic/extensions/rout_irr/src/rout_start.c
	modified:   vic/extensions/rout_rvic/src/rout_init.c

yixinmao committed:
Fix out soil moist bug (#551)

* Fixed a bug when outputting `OUT_SOIL_MOIST` and `OUT_SOIL_{LIQ/ICE}_FRAC`

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into develop

Commit:
rename atmos_data_struct-->force_data_struct

Commit:
eliminate two compiler warnings, minor refactor of read_soilparam function handle

Commit:
enable multiple output files (in time) per history stream

Commit:
Merge pull request #552 from jhamman/feature/rename_atmos

Feature/rename atmos

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into develop

Commit:
Merge pull request #554 from jhamman/feature/multifile_history_streams

Multifile history streams

wietsefranssen committed:
changed units of OUT_DISCHARGE to [m3 s-1]

Commit:
Check that the domain information in the parameters file matches the global domain info (#555)

* check that the domain information in the image driver parameters file matches the global domain (file)

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into develop

wietsefranssen committed:
fixed mpi_rank == VIC_MPI_ROOT

wietsefranssen committed:
little fix

wietsefranssen committed:
little fix

wietsefranssen committed:
placed everything on MPI_ROOT

Commit:
read/write state file and rpointer file in cesm driver (#556)

Commit:
Merge pull request #557 from wietsefranssen/fix/vic_mpi_root

fixed mpi_rank == VIC_MPI_ROOT

wietsefranssen committed:
Made RVIC working with MPI

wietsefranssen committed:
'check_alloc_status(dvar' should be 'check_alloc_status(dvar_filtered'. fixed now

wietsefranssen committed:
fixed hardcoded mpi path in Makefile

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

Commit:
Merge pull request #558 from wietsefranssen/fix/dvar_filtered

check_alloc_status

Commit:
Merge pull request #559 from wietsefranssen/fix/makefile_mpi

fixed hardcoded mpi path in Makefile

Commit:
Profiling: Feature/timing (#520)

* add timing module to image and classic drivers

* update release notes

* supress valgrind errors from non-vic libraries

wietsefranssen committed:
little fix

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

wietsefranssen committed:
tiny update

BramDr committed:
modified:   vic/drivers/image/src/vic_image.c
	modified:   vic/drivers/shared_image/src/vic_init_output.c
	modified:   vic/extensions/rout_irr/src/rout_run.c

wietsefranssen committed:
j and i index should be switched. Fixed now (#562)

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

wietsefranssen committed:
tiny update

wietsefranssen committed:
little update

Commit:
Cleanup image driver build (#565)

* image driver updates for building VIC on hyak, use nc-config to determine netcdf libraries

* allow env var specification of netcdf libs and mpicc in image makefile

yixinmao committed:
Add MPI test (#567)

* Add mpi multiprocessing test - test whether the flux and state results
using different number of processors are exactly identical for image
driver

* Add instruction on running VIC image driver using multiple processors to
the documentation

* Remove state output in mpi test - for debugging purpose

* Add back outputing state file in mpi test

* Add debug lines in VIC source code to debug failing MPI when writing
state file

* fixed bug of missing statesec in mpi global parameter file

* Remove some debugging lines in the VIC code for detecting MPI bug (bug
fixed)

Commit:
VIC 5 performance profiling (#560)

* add profiling tests and also print timing table from end of std out when profiling -- usefull when understanding the runtime along side the call graph

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into develop

Commit:
cleanup compiler warnings for intel builds

Commit:
fix memcpy void pointer warnings

Commit:
initialize iter_soil_energy

Commit:
update image driver run docs

Commit:
Fixed bug in calculating air density

Commit:
mpich 3.2 went missing, now using 3.1b1

Commit:
Merge pull request #568 from jhamman/fix/intel_warnings

Fix/intel warnings

wietsefranssen committed:
smaller/greater than sign switched (#561)

Commit:
Merge branch 'develop' into fix_bugs_AirDensity

Commit:
Merge pull request #569 from yixinmao/fix_bugs_AirDensity

Fix bug in air density calculation

Commit:
run uncrustify on all sources, fix some formating in timers

bartnijssen committed:
Merge pull request #572 from jhamman/feature/uncrustify_update2

run uncrustify on all sources, fix some formating in timers

yixinmao committed:
Fixed bugs related to ice heat capacity (#570)

* Fixed bugs related to ice heat capacity
    - In vic_run code, fixed bugs that confused volumetric heat capacity
      and specific heat capacity related to ice

* Combine the volumetric ice heat capacity (per volume of water
equivalent) into a single constant

Commit:
Fixed a bug related to VP - to be consistent with VIC4.2
    - When calculated VPD < 0, now resets VPD to zero, and VP to
      calculated saturated vapor pressure (this is consistent with
VIC4.2)

bartnijssen committed:
Merge pull request #573 from yixinmao/fix_bug_vpd

Fixed a bug related to VP - to be consistent with VIC4.2

dgergel committed:
Feature/add science testing (#544)

* added global parameter files for snotel forcings for vic 4.2 and 5.0

* remove hard-coded paths and made LAI_SRC the same for both versions

* updated LAI_SRC option

* cleaned up global parameter files for 4.2 and 5.0

* added global parameter files for Ameriflux simulations for 4.2 and 5.0

* Revert "added global parameter files for Ameriflux simulations for 4.2 and 5.0"

This reverts commit 60ca012bcdf1266ecbf60d4a4fb47f8bfbe5113c.

* add global parameter files for Ameriflux simulations

* add config file for science testing

* refactored science testing config file

* added science test plots

* added main science plots function and snotel plotting function

* finished adding plot_snotel_comparison function

* added string splitting function

* add constants file to science testing to help replicate VIC 4.2 results

* added fluxnet plot function and annual mean diurnal cycle section

* added monthly mean diurnal cycle section to plot fluxnet function

* add optional argument for science testing directory

* updated to only include one output file for both snow and energy balance output variables

* fixing up paths for science testing

* further fixing of paths

* updated global parameter files and minor bug fixes

* added matplotlib to required dependencies

* cleaned up global parameter files and updated precision for outvars in output files

* refactored science testing by separating reading functions and plotting functions

* remove print statement

* updated config file to include multi-level options

* removed warnings from requirements

* fix science test data directory check

* fixed typo on vic_parameters comparison to 42 file

* fix bug in science test directory check

* updated read_vic_ascii file and added output file nans check back in

* remove print statement

* fix bug in read_vic_ascii function

* fix travis build

* updated plotting functionality and formatting of vic output read functions

* further resolution of merge conflict

* update VIC 5 runs to include VIC 4.2 parameters (to account for difference in emissivity value)

* updated radiation output variable names

* typo in outputting net longwave radiation

* fix more typos in global parameter files for net shortwave and longwave output

* minor updates due to changes in VIC classic driver since test simulations were last run

* fixed pep8 issues as well as other formatting problems

* changed name of VIC 42 compatible parameters file

* remove old VIC 42 parameters file

* additional formatting changes and updates to the datetime indexing of the ascii vic output and snotel obs reader functions

* fix travis build

* updated vic 42 constants file

* fix small bug in read_snotel_swe_obs

yixinmao committed:
Fixed bugs related to volumetric heat capacity of water (#574)

* Fixed bugs related to volumetric heat capacity of water

* Added bug fix notes related to water heat capacity bug fix in
`func_canapy_energy_bal.c`

* Add PR number in ReleaseNotes bug fix

* Minor update of ReleaseNotes

dgergel committed:
updated paths to agree with directory structure of the VIC5_test_data ftp server (#577)

yixinmao committed:
Add test drivers match (#576)

* Add mpi multiprocessing test - test whether the flux and state results
using different number of processors are exactly identical for image
driver

* Add instruction on running VIC image driver using multiple processors to
the documentation

* Add cross-driver comparison test - half-way finished

* Turn on SNOW_BAND in cross-driver testing global param template file

* Fixed bugs related to ice heat capacity
    - In vic_run code, fixed bugs that confused volumetric heat capacity
      and specific heat capacity related to ice

* Combine the volumetric ice heat capacity (per volume of water
equivalent) into a single constant

* Finished driver comparison test

* Fixed a bug in run_tests.py

* Made system_tests.cfg consistent with develop branch

* Minor formatting

* Fixed a bug so that both single- and multi-driver tests are compatible
with system, science and example tests

* Fixed a bug in run_tests.py to be compatible with science tests

* Clean up test code - mostly related to handling cross-driver tests

* Removed an unused domain file from test directory

* Test code clean up; remove unused domain file under test/example

* Update travis build for classic driver and image driver

* Fixed a bug when calling run_example

* Fixed a bug related to drop_tests

Commit:
change units of image driver forcing VP to kPa from g/g (specific humidity).  See GH#418 for discussion

bartnijssen committed:
Merge pull request #584 from jhamman/feature/image_vp

Image Driver: SHUM-->VP

yixinmao committed:
Science test update (#583)

* Add mpi multiprocessing test - test whether the flux and state results
using different number of processors are exactly identical for image
driver

* Add instruction on running VIC image driver using multiple processors to
the documentation

* Add cross-driver comparison test - half-way finished

* Turn on SNOW_BAND in cross-driver testing global param template file

* Fixed bugs related to ice heat capacity
    - In vic_run code, fixed bugs that confused volumetric heat capacity
      and specific heat capacity related to ice

* Combine the volumetric ice heat capacity (per volume of water
equivalent) into a single constant

* Finished driver comparison test

* Fixed a bug in run_tests.py

* Made system_tests.cfg consistent with develop branch

* Minor formatting

* Fixed a bug so that both single- and multi-driver tests are compatible
with system, science and example tests

* Fixed a bug in run_tests.py to be compatible with science tests

* Clean up test code - mostly related to handling cross-driver tests

* Removed an unused domain file from test directory

* Test code clean up; remove unused domain file under test/example

* Update travis build for classic driver and image driver

* Fixed a bug when calling run_example

* Fixed a bug related to drop_tests

* Update science testing:
    - Updated paths in code and in global parameter template file
    - Update globap parameter template files for VIC4.2 and VIC5 so that
      the two versions take in consistent global options (although
VIC4.2 is run offline)

bartnijssen committed:
logging cleanup (see issue #526) (#581)

* logging cleanup (see issue #526)

* Removed function names from logging messages
* Removed newlines from the end of logging messages (added by log_*() functions)

* simplified unnecessary log_warn(); log_err() sequence

* One more newline removed

* Remove extraneous commas

* Removed extraneous error and warning in log_error()

Commit:
updates for rc2 tomorrow (#585)

* updates for rc2 tomorrow

* run uncrustify

Commit:
update docs for vic5

Commit:
doc fixes

Commit:
update docs for vic5 final release (#587)

* update docs for vic5

* doc fixes

bartnijssen committed:
Point DOI to latest version on zenodo

Commit:
Update image driver VP forcing name and remove sample global files
    - Updated VP forcing variable name in all image driver global
      parameter files (from `shum` to `vp`) to be consistent with image
driver VP unit change in the code
    - Remove sample global parameter files under `sample` directory -
      only keep a single sample global parameter file version under
`VIC_sample_data` repository

Commit:
Merge pull request #594 from yixinmao/update_forcing_image_VP

Update image driver VP forcing name and remove sample global files

Commit:
Updated vic_force in image driver - now input forcing units are
consistent with classic driver forcing units

Commit:
Merge pull request #595 from yixinmao/update_image_vic_force

Updated vic_force in image driver - now input forcing units are consistent with classic driver

dgergel committed:
added documentation for image driver forcing file

dgergel committed:
fix typo with describing number of processors

Commit:
Merge pull request #598 from dgergel/feature/fix_typo_run_image_driver

Feature/fix typo in run image driver

dgergel committed:
Merge branch 'develop' of github.com:UW-Hydro/VIC into feature/add_image_driver_forcing_documentation

dgergel committed:
updated documentation to specify calendar year chunking and global parameter file

dgergel committed:
updated forcing file name

dgergel committed:
updated ncdump example

Commit:
Merge pull request #597 from dgergel/feature/add_image_driver_forcing_documentation

added documentation for image driver forcing file

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into feature/vic5_docs

Commit:
documentation updates to all drivers and supporting information, pre VIC5 release

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

Commit:
add pull request and issue templates for github issues

Commit:
Merge pull request #603 from jhamman/feature/github_pr_template

Add templates for github issues and pull requests

Commit:
Merge branch 'develop' of github.com:UW-Hydro/VIC into feature/vic5_docs

Commit:
Merge pull request #600 from jhamman/feature/vic5_docs

updates to VIC5 docs

Commit:
Fixed a broken link in image driver documentation

Fixed a link to the parameter conversion notebook in the image driver documentation.

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

Commit:
Merge pull request #604 from hhelgason/develop

Fixed a broken link in image driver documentation

wietsefranssen committed:
Merge branch 'develop' of https://github.com/UW-Hydro/VIC into feature/routing_lohmann

BramDr committed:
Temporal save for new VIC update

BramDr committed:
Merge branch 'feature/routing_lohmann' of https://github.com/wietsefranssen/VIC into feature/rout_irr

BramDr committed:
Working version of the basic routing module (no irrigation)
